src/modules/kvs/kvs.c

```diff
@@ -2707,14 +2707,50 @@ static void process_args (struct kvs_ctx *ctx, int ac, char **av)
     }
 }
 
+static int checkpoint_get_version0 (flux_t *h, const char *value,
+                                    char *buf, size_t len)
+{
+    if (strlen (value) >= len) {
+        errno = EINVAL;
+        return -1;
+    }
+    strcpy (buf, value);
+    return 0;
+}
+
+static int checkpoint_get_version1 (flux_t *h, json_t *o,
```

Comment:
Drop unused `flux_t  *` parameter.

Reply:
Suggestion:  take string value not `json_t *` and do the `json_loads()` in this function.
Similarly, don't waste effort on errno here.